hadoop-tools/hadoop-aws/src/test/java17/org/apache/hadoop/fs/contract/s3a/ITestIcebergBulkDelete.java

```diff
@@ -0,0 +1,309 @@
+      final FileSystem fs = getFileSystem();
+      fs.mkdirs(path);
+      touch(fs, child);
+
+      LOG.info("Deleting path to directory");
+      if (useBulk) {
+        fileIO.deleteFiles(filename);
+      } else {
+        final BulkDeletionFailureException ex =
+            intercept(BulkDeletionFailureException.class, () ->
+                fileIO.deleteFiles(filename));
+        Assertions.assertThat(ex.numberFailedObjects())
+            .describedAs("Failure count in %s", ex)
+            .isEqualTo(1);
+      }
+      // Reported failure or not, the directory is still found
+      assertPathExists("directory was not deleted", path);
+    }
+  }
+
+  /**
+   * A directory is not deleted through the bulk delete API,
+   * it is through the classic single file delete.
+   * The assertions match this behavior.
+   * <p>
+   * Note that the semantics of FileSystem.delete(path, nonrecursive)
+   * have special handling of deleting an empty directory, where
+   * it is allowed (as with unix cli rm), so a child file
+   * is created to force stricter semantics.
+   */
+  @Test
+  public void testDeleteDirectoryDirect() throws Throwable {
+    //Path path = new Path(methodPath(), "../single");
+    Path path = methodPath();
+    try (HadoopFileIO fileIO = createFileIO()) {
+
+      // create a directory and a child underneath
+      Path child = new Path(path, "child");
+      final FileSystem fs = getFileSystem();
+
+      fs.mkdirs(path);
+      touch(fs, child);
+
+      LOG.info("Deleting path to directory via deleteFile");
+      intercept(RuntimeIOException.class, () ->
+      {
+        final String s = toString(path);
+        fileIO.deleteFile(s);
+        final FileStatus st = getFileStatusOrNull(fs, path);
+        return String.format("Expected failure deleting %s but none raised. Path status: %s",
+            path, st);
+      });
+    }
+  }
+
+  @Test
+  public void testDeleteManyFiles() throws Throwable {
+    Path path = methodPath();
+    final FileSystem fs = getFileSystem();
+    final List<Path> files = createFiles(fs, path, 1, DELETE_FILE_COUNT, 0);
+    try (HadoopFileIO fileIO = createFileIO()) {
+      fileIO.deleteFiles(stringList(files));
+      for (Path p : files) {
+        assertPathDoesNotExist("expected deletion", p);
+      }
+    }
+  }
+
+  /**
+   * Use a more complex filename.
+   * This validates that any conversions to URI/string
+   * when passing to an object store is correct.
+   */
+  @Test
+  public void testDeleteComplexFilename() throws Exception {
+    Path path = new Path(basePath, "child[=comple]x");
+    List<Path> paths = new ArrayList<>();
+    paths.add(path);
+    // bulk delete call doesn't verify if a path exists or not before deleting.
+    assertSuccessfulBulkDelete(bulkDelete_delete(getFileSystem(), basePath, paths));
+  }
+
+  public static List<String> stringList(List<Path> files) {
+    return files.stream().map(p -> toString(p)).collect(Collectors.toList());
```

Comment:
Can this be reduced to:

```
    return files.stream().map(Path::toString).collect(Collectors.toList());
```

...and then remove the static `toString` method below? It just seems to turn around and call `toString()` on the object.